proj-xcode/PowerAuth2/PowerAuthSDK.m

```
@@ -901,6 +903,85 @@ - (PowerAuthActivationStatus*) lastFetchedActivationStatus
     return status;
 }
 
+#pragma mark - Protocol upgrade
+
+- (id<PowerAuthOperationTask>) startProtocolUpgradeWithCorePassword:(PowerAuthCorePassword*)password
+                                                 withNewBiometryKek:(PowerAuthCoreData*)newBiometryKek
+                                                           callback:(void(^)(id status, NSError * error))callback
+{
+    [_lock lock];
+    
+    id<PowerAuthOperationTask> task = [_protocolUpgradeTask createChildTask:callback];
+    if (!task) {
+        PowerAuthCoreData * biometryKek = nil;
+        if (self.hasBiometryFactor) {
+            if (newBiometryKek) {
+                biometryKek = newBiometryKek;
+            } else {
+                biometryKek = [_sessionInterface readTaskWithSession:^PowerAuthCoreData* _Nullable(PowerAuthCoreSession* session, NSError** error) {
+                    return [PowerAuthCoreSession generateFactorKekForProtocolVersion:PowerAuthCoreProtocolVersion_V4 error:error];
```

`generateFactorKekForProtocolVersion` class method doesn't need session locking. The call is similar to static method call, with no `this` and therefore no data locking is needed.

Also, I would rather not swallow the error potentially reported from the function.